CONTRIBUTING.md

```diff
@@ -89,6 +92,60 @@ Use the test scripts provided in the `ci/` directory to run tests for each compo
 ./ci/test_cub.sh -cxx g++ -std 14 -arch "70;75;80-virtual"
 ```
 
+### Using CMake Presets
+
+[CMake Presets](https://cmake.org/cmake/help/latest/manual/cmake-presets.7.html) are a set of configurations defined in a JSON file that specify project-wide build details for CMake. They provide a standardized and sharable way to configure, build, and test projects across different platforms and development environments. Presets are available from CMake versions 3.19 and later.
+
+There are three kinds of Presets
+
+- Configure Presets: specify options for the `cmake` command,
+
+- Build Presets: specify options for the `cmake --build` command,
+
+- Test Presets: specify options for the `ctest` command.
+
+In CCCL we provide many presets to be used out of the box. You can find the complete list in our corresponding [CMakePresets.json](./CMakePresets.json) file.
```

Comment:
There's a better way than parsing the json manually. These commands can be used to get lists of the configure, build, and test presets. They strip out the `hidden` presets that aren't intended to be used directly, and provide a customizable description of each preset.

```
cmake --list-presets # Configure presets
cmake --build --list-presets # Build presets
ctest --list-presets # Test presets
```

While there is a lot of overlap, there may be differences between the configure, build, and test presets to support various testing workflows. 

The `dev` presets are intended as a base for general development while the others are useful for screening / replicating CI failures. The non-`dev` presets are invoked by the manual build/test scripts as needed.

The CI presets change from time to time as they are tied to our CI system. They may be renamed or removed. The dev presets are more stable.

Reply:
Ngl, that is much needed info. Will include it. Thanks.